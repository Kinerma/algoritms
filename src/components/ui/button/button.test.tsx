import {Button} from "./button";
import {render, screen, fireEvent} from "@testing-library/react";
import renderer from 'react-test-renderer'

describe('Button', () => {
    it('Отрисовка кнопки с текстом', function () {
        const button = renderer.create(<Button text='text' />).toJSON()
        expect(button).toMatchSnapshot()
    });
    it('Отрисовка кнопки без текста', function () {
        const button = renderer.create(<Button />).toJSON()
        expect(button).toMatchSnapshot()
    });
    it('Отрисовка заблокированной кнопки', function () {
        const button = renderer.create(<Button disabled />).toJSON()
        expect(button).toMatchSnapshot()
    });
    it('Отрисовка кнопки с индикацией загрузки', function () {
        const button = renderer.create(<Button isLoader />).toJSON()
        expect(button).toMatchSnapshot()
    });
    it('Проверяем корректность вызова колбека при клике на кнопку', function () {
        const onClick = jest.fn()
        render(<Button onClick={onClick} />)
        fireEvent.click(screen.getByRole('button'))
        expect(onClick).toHaveBeenCalled()
    });
})
