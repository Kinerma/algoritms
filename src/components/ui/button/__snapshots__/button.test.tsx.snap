// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Отрисовка заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Отрисовка кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
